diff --git a/node_modules/@sentry/react-native/.DS_Store b/node_modules/@sentry/react-native/.DS_Store
new file mode 100644
index 0000000..36d39ae
Binary files /dev/null and b/node_modules/@sentry/react-native/.DS_Store differ
diff --git a/node_modules/@sentry/react-native/ios/RNSentry.mm b/node_modules/@sentry/react-native/ios/RNSentry.mm
index 78b0a24..d5a75e3 100644
--- a/node_modules/@sentry/react-native/ios/RNSentry.mm
+++ b/node_modules/@sentry/react-native/ios/RNSentry.mm
@@ -311,6 +311,8 @@ RCT_EXPORT_METHOD(captureEnvelope:(NSArray * _Nonnull)bytes
     resolve(@YES);
 }
 
+#if SENTRY_HAS_UIKIT
+
 RCT_EXPORT_METHOD(captureScreenshot: (RCTPromiseResolveBlock)resolve
                   rejecter: (RCTPromiseRejectBlock)reject)
 {
@@ -354,6 +356,8 @@ RCT_EXPORT_METHOD(fetchViewHierarchy: (RCTPromiseResolveBlock)resolve
     resolve(viewHierarchy);
 }
 
+#endif
+
 RCT_EXPORT_METHOD(setUser:(NSDictionary *)userKeys
                   otherUserKeys:(NSDictionary *)userDataKeys
 )
